catalogue/webapp/components/ImageEndpointSearchResults/ImageEndpointSearchResults.tsx

```diff
@@ -71,6 +79,54 @@ const ImageEndpointSearchResults: FunctionComponent<Props> = ({
   const { isFullSupportBrowser } = useContext(AppContext);
   const [expandedImage, setExpandedImage] = useState<Image | undefined>();
   const [isActive, setIsActive] = useState(false);
+  const router = useRouter();
+
+  const getImage = async routerImageId => {
+    const apiUrl = `https://api.wellcomecollection.org/catalogue/v2/images/${routerImageId}`;
+    const image: Image = await fetch(apiUrl).then(res => res.json());
+
+    if (image) {
+      setExpandedImage(image);
+      setIsActive(true);
+    }
+  };
+
+  const encodeDecodeImageIdInUrl = (mode: 'encode' | 'decode') => {
+    const searchTest = window.location.search;
+    const search = searchTest?.slice(1) || '';
+    const params = parse(search);
+
+    if (mode === 'encode') {
+      params.imageId = expandedImage?.id;
+    }
+    if (mode === 'decode') {
+      delete params.imageId;
+    }
+
+    router.push(`${router.pathname}?${stringify(params)}`, undefined, {
+      shallow: true,
+    });
+  };
```

Comment:
I'm not sure that `decode` is the right term for this - it looks like it's removing the parameter from the URL. Let's have 2 separate functions for the 2 actions.